write-fonts/src/tables/gvar.rs

```diff
@@ -688,4 +760,158 @@ mod tests {
+    }
+
+    // comparing our behaviour against what we know fonttools does.
+    #[test]
+    #[allow(non_snake_case)]
+    fn oswald_Lcaron() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let d1 = GlyphDeltas::new(
+            Tuple::new(vec![F2Dot14::from_f32(-1.0), F2Dot14::from_f32(-1.0)]),
+            vec![
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::required(35, 0),
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::required(-24, 0),
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::optional(0, 0),
+            ],
+            None,
+        );
+
+        let d1_sparse = GlyphDeltas::build_sparse_data(&d1.deltas);
+
+        assert_eq!(
+            d1_sparse
+                .private_point_numbers
+                .clone()
+                .unwrap()
+                .compute_size(),
+            4
+        );
+        assert_eq!(d1_sparse.x_deltas.compute_size(), 3);
+        assert_eq!(d1_sparse.y_deltas.compute_size(), 1);
+
+        let d1_dense = GlyphDeltas::build_non_sparse_data(&d1.deltas);
+
+        assert_eq!(d1_dense.x_deltas.compute_size(), 6);
+        assert_eq!(d1_dense.y_deltas.compute_size(), 1);
+
+        assert_eq!(d1_sparse.compute_size(), d1_dense.compute_size());
+
+        let d2 = GlyphDeltas::new(
+            Tuple::new(vec![F2Dot14::from_f32(1.0), F2Dot14::from_f32(1.0)]),
+            vec![
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::required(26, 15),
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::required(46, 0),
+                GlyphDelta::optional(0, 0),
+                GlyphDelta::optional(0, 0),
+            ],
+            None,
+        );
+        let d2_sparse = GlyphDeltas::build_sparse_data(&d2.deltas);
+
+        assert_eq!(
+            d2_sparse
+                .private_point_numbers
+                .as_ref()
+                .unwrap()
+                .compute_size(),
+            4
+        );
+        assert_eq!(d2_sparse.x_deltas.compute_size(), 3);
+        assert_eq!(d2_sparse.y_deltas.compute_size(), 3,);
+
+        let d2_dense = GlyphDeltas::build_non_sparse_data(&d2.deltas);
+
+        assert_eq!(d2_dense.x_deltas.compute_size(), 6);
+        assert_eq!(d2_dense.y_deltas.compute_size(), 4);
+
+        assert!(d2_sparse.compute_size() < d2_dense.compute_size());
+
+        let tups = HashMap::new();
+        let variations = GlyphVariations::new(GlyphId::new(0), vec![d1, d2]);
+        assert!(variations.compute_shared_points().is_none());
+        let built = variations.build(&tups);
+        assert_eq!(
+            built.per_tuple_data[0].private_point_numbers,
+            Some(PackedPointNumbers::All)
+        );
+        assert_eq!(
+            built.per_tuple_data[1].private_point_numbers,
+            Some(PackedPointNumbers::Some(vec![1, 3]))
+        );
+    }
```

Comment:
This looks suspiciously like many tests shoved into one because they share test data. Make a helper to provide the data and split up the tests?

Reply:
yes fair. this test is noisy because I was using it to debug something. The only thing we really care about here is the very last assertions, that we use all the points for the first tuple but sparse points for the second; all the intermediate stuff is just leftovers from my debugging, can clean that up.